File: ProtocolTestManager/PTMService/PTMService/ClientApp/src/model/TestCaseResult.ts
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

export type TestCaseState = 'NotRun' | 'Running' | 'Passed' | 'Failed' | 'Inconclusive'

export interface TestCaseOverview {
  FullName: string
  State: TestCaseState
}

export interface TestCaseResult {
  Overview: TestCaseOverview
  StartTime: string
  EndTime: string
  Output: string
}

File: ProtocolTestManager/PTMService/Common/Types/TestCaseResult.cs
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Microsoft.Protocols.TestManager.PTMService.Common.Types
{
    public class TestCaseResult
    {
        public TestCaseOverview Overview { get; set; }

        public string StartTime { get; set; }

        public string EndTime { get; set; }

        public string Output { get; set; }
    }
}
